From d59918e0b6e49fb68da5f11a9f17826a1de81b12 Mon Sep 17 00:00:00 2001
From: 
Date: Mon, 8 Apr 2019 18:12:43 -0500
Subject: [PATCH 3/3] Portability fixes:

- Use fcntl.h for O_RDONLY and friends.
- Only use mallinfo with glibc.

---
 lib/mm/memlock.c                 | 2 +-
 libdaemon/server/daemon-server.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/mm/memlock.c b/lib/mm/memlock.c
index c8e6ef1..81b74fc 100644
--- a/lib/mm/memlock.c
+++ b/lib/mm/memlock.c
@@ -159,7 +159,7 @@ static void _touch_memory(void *mem, size_t size)
 
 static void _allocate_memory(void)
 {
-#ifndef VALGRIND_POOL
+#if !defined(VALGRIND_POOL) && defined(__GLIBC__)
 	void *stack_mem;
 	struct rlimit limit;
 	int i, area = 0, missing = _size_malloc_tmp, max_areas = 32, hblks;
diff --git a/libdaemon/server/daemon-server.c b/libdaemon/server/daemon-server.c
index 3b19883..4a38de9 100644
--- a/libdaemon/server/daemon-server.c
+++ b/libdaemon/server/daemon-server.c
@@ -18,6 +18,7 @@
 #include "daemon-server.h"
 #include "daemon-log.h"
 
+#include <fcntl.h>
 #include <dlfcn.h>
 #include <errno.h>
 #include <pthread.h>
-- 
2.21.0

